core/lib/rom/registry.rb

```diff
@@ -12,12 +12,39 @@ class Registry
     include Enumerable
     include Dry::Equalizer(:elements)
 
-    param :elements, default: -> { Hash.new }
+    param :elements
+
+    option :cache, reader: true, default: -> { Cache.new }
+
+    def self.new(*args)
```

Comment:
I wonder if we really need this, dry-initializer should work fine with inheritance, so perhaps we might remove `param :elements` from the descendants instead. Not sure though